sda_download/sda_download.go

```diff
@@ -106,29 +105,56 @@ func SdaDownload(args []string) error {
 		return err
 	}
 
-	// Loop through the files and download them
-	for _, filePath := range files {
-		fileIDURL, err := getFileIDURL(*URL, config.AccessToken, *datasetID, filePath)
+	// Check if dataset flag is set
+	// If it is, download all files in the dataset
+	// If it is not, download the files that are provided
+	if *datasetdownload {
+		err = datasetCase(config.AccessToken)
+		if err != nil {
+			return err
+		}
+	} else {
+		err = fileCase(config.AccessToken)
 		if err != nil {
 			return err
 		}
+	}
+
+	return nil
+}
 
-		// Check if the file path contains a userID and if it does,
-		// do not keep it in the file path
-		filePathSplit := strings.Split(filePath, "/")
-		if strings.Contains(filePathSplit[0], "_") {
-			_, err := mail.ParseAddress(strings.ReplaceAll(filePathSplit[0], "_", "@"))
-			if err == nil {
-				filePath = strings.Join(filePathSplit[1:], "/")
-			}
+func datasetCase(token string) error {
+	fmt.Println("No files provided, downloading all files in the dataset")
```

Comment:
This message is a bit misleading, when the command is aimed for downloading the whole dataset, it doesn't matter if files are provided or not. 
Maybe just change it to `Downloading all files in the dataset`

Reply:
You are right, I have changed it